### PR TITLE
Ret 2058

### PR DESCRIPTION
**Pull request by robin-ieong**
https://tools.hmcts.net/jira/browse/RET-2058

### Change description ###
A Respondent has submitted an ET3 Form which has been vetted and accepted by case workers. Now all parties need to be told the form has been received and accepted, and a copy of the ET3 form must be sent to the Claimant.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/ET3NotificationControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ExtendWith(SpringExtension.class)
+@WebMvcTest({ET3NotificationController.class, JsonMapper.class})
+class ET3NotificationControllerTest {
+    private static final String AUTH_TOKEN = "Bearer eyJhbGJbpjciOiJIUzI1NiJ9";
+    private static final String ET3_NOTIFICATION_ENGPOINT = "/et3Notification";
```
Comment (jsinghm13) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/ET3NotificationControllerTest.java:37]: Need to fix the typo, should be ET3_NOTIFICATION_**ENDPOINT**

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/ET3NotificationControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        ccdRequest = CCDRequestBuilder.builder().withCaseData(caseData).build();
+    }
+
+    @Test
+    void et3Notification() throws Exception {
+        when(verifyTokenService.verifyTokenSignature(AUTH_TOKEN)).thenReturn(true);
+        when(servingHelper.generateOtherTypeDocumentLink(anyList())).thenReturn("expectedDocumentName");
+        when(servingHelper.generateEmailLinkToAcas(any(), anyBoolean())).thenReturn("expectedLink");
+        when(servingHelper.generateClaimantAndRespondentAddress(any())).thenReturn("expectedAddresses");
+        mvc.perform(post(ET3_NOTIFICATION_ENGPOINT)
```
Comment (jsinghm13) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/ET3NotificationControllerTest.java:62]: Fix typo - ET3_NOTIFICATION_ENGPOINT

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/ET3NotificationControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+            .andExpect(jsonPath("$.warnings", nullValue()));
+        verify(servingHelper, times(1)).generateOtherTypeDocumentLink(anyList());
+        verify(servingHelper, times(1)).generateEmailLinkToAcas(any(), anyBoolean());
+        verify(servingHelper, times(1)).generateClaimantAndRespondentAddress(any());
+    }
+
+    @Test
+    void et3NotificationForbidden() throws Exception {
+        when(verifyTokenService.verifyTokenSignature(AUTH_TOKEN)).thenReturn(false);
+        mvc.perform(post(ET3_NOTIFICATION_ENGPOINT)
```
Comment (jsinghm13) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/ET3NotificationControllerTest.java:80]: Fix typo - ET3_NOTIFICATION_ENGPOINT

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/ServingHelperTest.java [hunk 4]
```diff
@@ -53,18 +54,23 @@ void generateClaimantAndRespondentAddress() {
...
             + "%20attached%20to%20this%20email.%0D%0A%0D%0A";
+        String expectedEt3EmailLinkToAcas = "mailto:ET3@acas.org.uk?subject=2120001/2019"
+            + "&body=Parties%20in%20claim%3A%20Doris%20Johnson%20vs%20Antonio%20Vazquez%2C%20Juan%20Garcia%0D%0A"
+            + "Case%20reference%20number%3A%202120001/2019%0D%0A%0D%0ADear%20Acas%2C%0D%0A%0D%0AThe%20tribunal%20"
+            + "has%20completed%20ET3%20notifications%20to%20the%20relevant%20parties.%0D%0A%0D%0AThe%20documents%20we"
+            + "%20sent%20are%20attached%20to%20this%20email.%0D%0A%0D%0A";
         CaseData caseData = caseDetails.getCaseData();
-        assertThat(ET1ServingService.generateEmailLinkToAcas(caseData), is(expectedEmailLinkToAcas));
+        assertThat(ServingHelper.generateEmailLinkToAcas(caseData, false), is(expectedEt1EmailLinkToAcas));
     }
```
Comment (jsinghm13) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/ServingHelperTest.java:75]: Need to test the expectedEt3EmailLinkToAcas as well

**Review by jsinghm13 — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/ServingHelper.java [hunk 5]
```diff
@@ -8,30 +8,44 @@
 
 import java.text.MessageFormat;
 import java.util.List;
+import java.util.stream.Collectors;
 
 @Service
-public class ET1ServingService {
+public class ServingHelper {
```
Comment (hsjhita) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/ServingHelper.java:14]: ServingService?

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/ET3NotificationControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    @MockBean
+    private ServingHelper servingHelper;
+    @Autowired
+    private MockMvc mvc;
+    @Autowired
+    private JsonMapper jsonMapper;
+
+    @BeforeEach
+    void setUp() {
+        var caseData = new CaseData();
```
Comment (hsjhita) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/ET3NotificationControllerTest.java:51]: remember - trying to cut down on the usage of var

**Review by hsjhita — APPROVED** [review]

**Pull request merged**